### ISSUE TITLE
Incorrect Model when JSON response contains "property_name": []

### ISSUE THREAD

**Issue by yched** [chargebee/chargebee-php#63]
### Description of the Bug

For model properties containing an array of sub-models, 
(for instance $subscription->coupons,  or $invoice->appliedCredits),

in cases where the property is returned as `"property_name": []` in the API JSON response,
then the Model instanciated is incorrect : 
$invoice->appliedCredits should be either null or an array of InvoiceAppliedCredit objects,
but instead it is directly a InvoiceAppliedCredit object with all null properties

### Steps to reproduce

Exemple of broken behavior :
Case of an invoice without applied credits
```
$invoice = \ChargeBee\ChargeBee\Models\Invoice::retrieve('the_invoice_id')->invoice();
// appliedCredits is not null, and is not an array of InvoiceAppliedCredit objects
$invoice->appliedCredits === null   // --> false
is_array($invoice->appliedCredits)   // --> false
// instead, it is directly a (single) InvoiceAppliedCredit object with all null properties
$i->appliedCredits instanceof \ChargeBee\ChargeBee\Models\InvoiceAppliedCredit    // --> true
$i->appliedCredits->cnId     // --> null
```

That is because the JSON result for `api/v2/invoices/the_invoice_id` contains : 
`"applied_credits": []`

This then goes into the Model::_load() method, 
and the [] is mistakenly handled as a "hash" array
(isHash([]) returns true, while it should return false)

```
foreach ($this->_values as $k => $v) {
    // $k is 'applied_credits', $v is []
    if ($this->isHash($v) && array_key_exists($k, $this->_subTypes)) {
        // We go here : new InvoiceAppliedCredit(|])
        $setVal = new $this->_subTypes[$k]($v);        
    } elseif (is_array($v) && array_key_exists($k, $this->_subTypes)) {
        // We should have gone here instead
        foreach ($v as $stV) {
            (... add new InvoiceAppliedCredit($stV)  ... )
        }
    }
}
```

### Expected Behavior

Exemple of correct behavior : 
Case of a subscription without coupons

```
$subscription = \ChargeBee\ChargeBee\Models\Subscription::retrieve('the_subscription_id')->subscription();
// coupons is null
$subscription->coupons === null // --> true
```

That is because the JSON result for `api/v2/subscriptions/the_subscription_id` omits the "coupons" property if it is empty

### Operating System

Linux

### Language version

Php 8.1.30

### Library version

v3.37.0

**Comment by cb-alish** [comment]
Hi @yched, the issue should be resolved in v3.37.1. Please give it a try and let us know if it persists. We'll keep this open until your confirmation.

**Comment by yched** [comment]
Hi @cb-alish, 
yes this is fixed with v3.37.1 :ok_hand: 

Thanks for the wuick response ! :+1:
